codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeDeserVisitor.java

```diff
@@ -67,32 +69,78 @@ public XmlShapeDeserVisitor(GenerationContext context, Predicate<MemberShape> me
         return Collections.singletonMap("decoder", "*smithydecoding.XMLNodeDecoder");
     }
 
-    private XmlMemberDeserVisitor getMemberDeserVisitor(MemberShape member, String dataDest, Predicate<Shape> filter) {
-        // Get the timestamp format to be used, defaulting to epoch seconds.
+    private XmlMemberDeserVisitor getMemberDeserVisitor(MemberShape member, String dataDest, Boolean isXmlAttibuteMember) {
+        // Get the timestamp format to be used, defaulting to rfc 3339 date-time format.
         TimestampFormatTrait.Format format = member.getMemberTrait(getContext().getModel(), TimestampFormatTrait.class)
                 .map(TimestampFormatTrait::getFormat).orElse(DEFAULT_TIMESTAMP_FORMAT);
-
         // Check if FlattenedTrait  is applied
         Boolean isFlattened = member.hasTrait(XmlFlattenedTrait.ID);
-
-        return new XmlMemberDeserVisitor(getContext(), dataDest, format, filter, isFlattened);
+        return new XmlMemberDeserVisitor(getContext(), dataDest, format, isXmlAttibuteMember, isFlattened);
     }
 
-    @Override
-    protected void deserializeCollection(GenerationContext context, CollectionShape shape) {
+    // generates code to define and initialize output variable for an aggregate shape
+    private void generatesIntializerForOutputVariable(GenerationContext context, Shape shape) {
         GoWriter writer = context.getWriter();
-        SymbolProvider symbolProvider = context.getSymbolProvider();
-        Symbol symbol = symbolProvider.toSymbol(shape);
+        Symbol shapeSymbol = context.getSymbolProvider().toSymbol(shape);
 
-        // Initialize the value now that the start stub has verified that there's something there.
-        writer.write("var sv $P", symbol);
+        writer.write("var sv $P", shapeSymbol);
         writer.openBlock("if *v == nil {", "", () -> {
-            writer.write("sv = make($P, 0)", symbol);
+            if (shape.isStructureShape()) {
+                writer.write("sv = &$T{}", shapeSymbol);
+            } else {
+                writer.write("sv = make($P, 0)", shapeSymbol);
+            }
             writer.openBlock("} else {", "}", () -> {
                 writer.write("sv = *v");
             });
         });
+    }
+
+    @Override
+    public Void mapShape(MapShape shape) {
+        generateDeserFunction(shape, (c, s) -> deserializeMap(c, s.asMapShape().get()));
```

Comment:
You should call the super here instead so you pick up any changes that happen there. Same for the others